src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilderMetricsLabelTest.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.jenkins.plugins.k8sengine;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+
+import hudson.FilePath;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.yaml.snakeyaml.Yaml;
+
+/** Tests the Kubernetes metrics label behaviors within {@link KubernetesEngineBuilder}. */
+@RunWith(MockitoJUnitRunner.class)
+public class KubernetesEngineBuilderMetricsLabelTest {
```

Comment:
Question: The other test class was getting pretty long so I'm glad you separated this out here. Do you think we'd want to split the other class up as well? Either split by different UI field (project ID, zone, cluster name) or descriptor method type (doCheck... vs doFill...Items)?

Reply:
Yeah I was thinking the same thing. Perhaps separated by the behavior set that's being tested like with the MetricsLabelTest. I'll add a TODO.